package_bag/routines.py

```diff
@@ -21,29 +21,26 @@ class BagDiscoverer(object):
     """
 
     def run(self):
-        processed = []
-        unprocessed = self.discover_bags(settings.SRC_DIR)
-        for bag in unprocessed:
-            try:
-                bag_id = self.unpack_rename(bag, settings.TMP_DIR)
-                bag_path = join(settings.TMP_DIR, bag_id)
-                validate(bag_path)
-                bag_data = self.validate_metadata(bag_path)
-                new_bag = Bag.objects.create(
-                    original_bag_name=bag,
-                    bag_identifier=bag_id,
-                    bag_path=bag_path)
-                for key in ["Origin", "Rights-ID", "End-Date"]:
-                    setattr(new_bag, key.lower().replace("-", "_"), bag_data.get(key))
-                new_bag.save()
-                processed.append(bag_id)
-            except Exception as e:
-                remove_file_or_dir(bag_path)
-                print(e)
+        bag = self.discover_bags(settings.SRC_DIR)[0]
```

Comment:
I would suggest modifying the `discover_bags` method so it just returns one item rather than a list. You'll also probably want to call it something else like `discover_next_bag`